### PR TITLE
feat(kit): add conditional entitlement refresh

### PR DESCRIPTION
**Pull request by hyochan**
## Summary

- add bounded, user-scoped subscription evaluation snapshots for hosted IAPKit status and entitlement reads
- support API-key-, route-, user-, and content-scoped weak ETags with conditional `304 Not Modified` responses
- document persistent local caching and cold-start, stale-foreground, or explicit refresh as the safe replacement for removed outbound event streams

### Hosted subscription refresh

- add an indexed `(projectId, userId, updatedAt)` lookup with a 201-row overflow probe and a fail-closed 200-row public contract
- return only stored-state `Active`/`InGracePeriod` candidates plus one latest fallback for Fly-side evaluation
- evaluate `expiresAt` against the Fly server clock on every HTTP request
- keep polling read-only and preserve existing status and entitlement response bodies for unconditional clients

### HTTP and cache safety

- isolate weak ETags by API key, route, user, and evaluated response content
- return `Vary: Authorization` and private cache directives for publishable-key responses
- keep secret/admin responses `private, no-store` without ETags
- preserve bounded rate limiting and `Retry-After` behavior

### Documentation

- add persistent project-and-user cache examples that coalesce refreshes and reject invalid or future cache timestamps
- recommend direct cold-start, stale-foreground, and explicit refreshes instead of foreground-wide polling
- document rollout ordering, cost characteristics, the legacy direct-Convex residual, and the permanent prohibition on IAPKit-to-SDK SSE/WebSockets/push relays/long polling

## Rollout

Deploy the additive Convex schema/query before the Fly server. Existing SDK helpers remain unconditional body-only reads, so this hosted-only change does not require a native or framework package version bump.

## Test plan

- [x] Kit Prettier, TypeScript, Convex typecheck, and ESLint
- [x] Kit test suite: 72 files / 858 tests
- [x] Kit production build, compiled server, and browser/API smoke probes
- [x] Docs Prettier, typecheck, lint, and production build
- [x] Documentation, SDK parity, deprecation, and release-state audits
- [x] Removed outbound stream routes and legacy key-in-path variants return `404`
- [x] `git diff --check`

## Preview

A visual recording is not applicable because this is a hosted HTTP/Convex contract and documentation change with no new interactive UI. The compiled-server smoke probes and regression tests above cover the changed surface.

## Summary by CodeRabbit

* **New Features**
  * Added user-scoped subscription status and entitlements endpoints with conditional snapshot refresh via `ETag`/`If-None-Match` and `304` reuse.
  * Implemented bounded subscription snapshot evaluation with “fail closed” on overflow.
  * Added a hosted refresh flow replacement that uses polling (not app-facing event streams), including `429` `Retry-After` guidance and stale-age/offline behavior.
* **Documentation**
  * Expanded SSE replacement and refresh/caching guidance, plus updated API reference and quickstart/release notes.
  * Updated integration version snippets and timestamps.
* **Tests**
  * Extended API and snapshot evaluation tests, including limit, `ETag`, and caching edge cases.

### CODE REVIEW

File: packages/kit/server/api/v1/subscriptions.ts [hunk 1]
```diff
@@ -457,6 +492,136 @@ function isNonBlankString(value: unknown): value is string {
...
+  active: boolean;
+  subscription: SubscriptionSnapshotRow | null;
+} {
+  const activeSubscriptions = snapshot.candidates.filter((subscription) =>
+    isActiveSubscription(subscription, now),
+  );
+  return {
+    active: activeSubscriptions.length > 0,
+    subscription: activeSubscriptions[0] ?? snapshot.fallback,
+  };
```
[the remark this answers is not in the document]
Comment (hyochan) [reply packages/kit/server/api/v1/subscriptions.ts:515]: Fixed in 1413729b. The evaluation snapshot now carries the Convex creation timestamp only as internal tie-break metadata, Fly explicitly selects by updatedAt and then creation time, and the field is stripped before status/entitlement responses and ETag generation. I also added a regression test for equal-updatedAt rows.

**Comment by hyochan** [comment]
@coderabbitai review

File: packages/kit/src/pages/docs/sections/api.tsx [hunk 2]
```diff
@@ -196,6 +196,187 @@ export default function ApiReferencePage() {
...
+  }
+
+  if (response.status === 304 && cached) {
+    const refreshed = { ...cached, checkedAt: Date.now() };
+    await persistSnapshot(refreshed);
+    return refreshed.snapshot;
+  }
+  if (response.status === 429) {
+    scheduleRetry(response.headers.get("Retry-After"));
+    if (cached && canUseCachedSnapshot()) return cached.snapshot;
```
[the remark this answers is not in the document]
Comment (hyochan) [reply packages/kit/src/pages/docs/sections/api.tsx]: Fixed in 484d0138. Both hosted and public documentation samples now scope ETag reuse, 304 handling, offline fallback, and 429 fallback to a cache whose snapshot userId matches the requested user.

**Comment by hyochan** [comment]
@coderabbitai review

**Comment by hyochan** [comment]
@coderabbitai review
